src/overlays/actors/ovl_En_Ruppecrow/z_en_ruppecrow.c

```diff
@@ -81,62 +112,569 @@ static DamageTable D_80BE39EC = {
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 12.0f, 25.0f, 50.0f, 0x07);
+
+    return true;
+}
+
+s32 EnRuppecrow_ReachedNextPoint_Clockwise(EnRuppecrow* this, Path* path, s32 index) {
+    Vec3s* points;
+    s32 pathCount;
+    s32 pathIndex;
+    s32 result;
+    f32 y;
+    f32 x;
+    Vec3f vec;
+    Vec3f point;
+
+    pathIndex = index;
+    points = Lib_SegmentedToVirtual(path->points);
+    pathCount = path->count;
+    result = false;
+
+    Math_Vec3s_ToVec3f(&point, &points[pathIndex]);
+
+    if (pathIndex == 0) {
+        y = points[1].x - points[0].x;
+        x = points[1].z - points[0].z;
+    } else {
+        if (pathIndex == (pathCount - 1)) {
+            y = points[pathCount - 1].x - points[pathCount - 2].x;
+            x = points[pathCount - 1].z - points[pathCount - 2].z;
+        } else {
+            y = points[pathIndex + 1].x - points[pathIndex - 1].x;
+            x = points[pathIndex + 1].z - points[pathIndex - 1].z;
+        }
+    }
+
+    func_8017B7F8(&point, (func_80086B30(y, x) * 10430.378f), &vec.z, &vec.y, &vec.x);
+    if (((this->actor.world.pos.x * vec.z) + (vec.y * this->actor.world.pos.z) + vec.x) > 0.0f) {
+        result = true;
+    }
+
+    return result;
+}
+
+s32 EnRuppecrow_ReachedNextPoint_CounterClockwise(EnRuppecrow* this, Path* path, s32 index) {
+    Vec3s* points;
+    s32 pathCount;
+    s32 pathIndex;
+    s32 result;
+    f32 y;
+    f32 x;
+    Vec3f vec;
+    Vec3f point;
+
+    pathIndex = index;
+    points = Lib_SegmentedToVirtual(path->points);
+    pathCount = path->count;
+    result = false;
+
+    Math_Vec3s_ToVec3f(&point, &points[pathIndex]);
+
+    if (pathIndex == 0) {
+        y = points[0].x - points[1].x;
+        x = points[0].z - points[1].z;
+    } else {
+        if (pathIndex == (pathCount - 1)) {
+            y = points[pathCount - 2].x - points[pathCount - 1].x;
+            x = points[pathCount - 2].z - points[pathCount - 1].z;
+        } else {
+            y = points[pathIndex - 1].x - points[pathIndex + 1].x;
+            x = points[pathIndex - 1].z - points[pathIndex + 1].z;
+        }
+    }
+
+    func_8017B7F8(&point, (func_80086B30(y, x) * 10430.378f), &vec.z, &vec.y, &vec.x);
+    if (((this->actor.world.pos.x * vec.z) + (vec.y * this->actor.world.pos.z) + vec.x) > 0.0f) {
+        result = true;
+    }
+
+    return result;
+}
+
+f32 EnRuppecrow_GetOrientationToNextPoint(Path* path, s32 arg1, PosRot* world, Vec3s* orientation) {
+    Vec3s* points;
+    Vec3f point;
+
+    if (path != NULL) {
+        points = Lib_SegmentedToVirtual(path->points);
+        points = &points[arg1];
+        point.x = points->x;
+        point.y = points->y;
+        point.z = points->z;
+    }
+
+    orientation->y = Math_Vec3f_Yaw(&world->pos, &point);
+    orientation->x = Math_Vec3f_Pitch(&world->pos, &point);
+
+    return point.y - world->pos.y;
+}
+
+void func_80BE2728(EnRuppecrow* this, GlobalContext* globalCtx) {
```

Comment:
Seems like this function is about shattering the ice effect, although I'm not sure what `func_800BF7CC` does. It's in `z_actor`, so @AngheloAlf probably has some idea?

Reply:
Yeah, this function spawns ice effects
<https://github.com/Decompollaborate/mm/blob/e4dc5e81711bd829613f2878846cd80eccada0b4/src/code/z_actor.c#L4684>

Reply:
So something like `EnRuppecrow_ShatterIce` or `BreakIce`, I suppose.

Reply:
Agreed.